### ISSUE TITLE
Changes of the `getTransactionReceipt` method response interface in v4.x.x

### ISSUE THREAD

**Issue by retroboydev** [0xs34n/starknet.js#296]
I'm not very good at typescript, but I will try to explain.

Here's an example of the code:

```
import { Provider } from "starknet";

const provider = new Provider({
    "baseUrl": "https://alpha4.starknet.io",
    "chainId": "0x534e5f474f45524c49",
    "feederGatewayUrl": "https://alpha4.starknet.io/feeder_gateway",
    "gatewayUrl": "https://alpha4.starknet.io/gateway"
}); // window.starknet.provider

provider
  .getTransactionReceipt(hash)
  .then(receipt => {
    result = {
      blockHash: receipt.block_hash, // not available in the interface anymore
      blockNumber: Number(receipt.block_number), // not available in the interface anymore
      status: receipt.status,
      transactionHash: receipt.transaction_hash,
      transactionIndex: Number(receipt.transaction_index) // not available in the interface anymore
    }
    dispatch(updateTransaction(result))
  })
```

Here's [the interface](https://github.com/0xs34n/starknet.js/blob/v3.18.2/src/types/api.ts#L257) of the response of the `.gettransactionReceipt` method in v 3.18.x:

```
export type SuccessfulTransactionReceiptResponse = {
  status: Status;
  transaction_hash: string;
  transaction_index: number;
  block_hash: string;
  block_number: BlockNumber;
  l2_to_l1_messages: string[];
  events: string[];
  actual_fee: string;
  execution_resources: ExecutionResources;
};
```

However, in version v4.x.x that interface was replaced with [this one](https://github.com/0xs34n/starknet.js/blob/efee0ac47b380eb2b1d187a7678a2f5ed068f598/src/types/provider.ts#L83):

```
export interface InvokeTransactionReceiptResponse extends CommonTransactionReceiptResponse {
  messages_sent: Array<MessageToL1>;
  events: Array<Event>;
  l1_origin_message?: MessageToL2;
}
```
I'm not sure if this is fine, since we still receive the same amount of data with the same fields from the same endpoint, and these fields are still available in the response. But now we have a typescript error about these fields since they are not available in the interface anymore. 

![image](https://user-images.githubusercontent.com/109076629/185164576-420f147b-9839-46bc-9983-56f28afdaa3a.png)

**Comment by ivpavici** [comment]
Hi! These are missing here from the rpc, and is a bug:
https://github.com/0xs34n/starknet.js/blob/efee0ac47b380eb2b1d187a7678a2f5ed068f598/src/types/api/rpc.ts#L55

But still, a question from us: do you want to use the rpc or the sequencer provider?

**Comment by retroboydev** [comment]
@ivpavici thank you for your response.

I've been working with starknet not so far and I'm not sure that I clearly understand the difference between them yet. We receive the provider object from `window.starknet.provider` (it may be created by ArgentX or Bravoos) and provide it to your constructor function:
```
import { Provider } from "starknet";
const provider = new Provider(window.starknet.provider)
```
Not sure what type of provider it is. As far as I can see in your code, in this case, you don't create the PRC or the Sequencer provider and just use the one we provided instead.

Please let me know if I understood your question wrong.

**Comment by tabaktoni** [comment]
Can you show log of window.starknet.provider

> @ivpavici thank you for your response.
> 
> I've been working with starknet not so far and I'm not sure that I clearly understand the difference between them yet. We receive the provider object from `window.starknet.provider` (it may be created by ArgentX or Bravoos) and provide it to your constructor function:
> 
> ```
> import { Provider } from "starknet";
> const provider = new Provider(window.starknet.provider)
> ```
> 
> Not sure what type of provider it is. As far as I can see in your code, in this case, you don't create the PRC or the Sequencer provider and just use the one we provided instead.
> 
> Please let me know if I understood your question wrong.

**Comment by ivpavici** [comment]
This one could possibly be fixed with the new version [4.6.0](https://github.com/0xs34n/starknet.js/releases/tag/v4.6.0)
@retroboydev can you try ?

**Comment by retroboydev** [comment]
@ivpavici sorry, no changes
![image](https://user-images.githubusercontent.com/109076629/190893173-86ab3e4f-4f42-43de-b81d-011e1af83e58.png)
![image](https://user-images.githubusercontent.com/109076629/190893207-4ec2a4d3-67bc-4ef5-98ef-1fed13ebaeb8.png)
